cogsmisc/customization.py

```diff
@@ -316,24 +321,50 @@ def _can_edit_servaliases(ctx):
            checks.author_is_owner(ctx)
 
 
-async def _alias_before_edit(ctx, name=None):
+async def _alias_before_edit(ctx, name=None, delete=False):
     if name and name in ctx.bot.all_commands:
         raise InvalidArgument(f"`{name}` is already a builtin command. Try another name.")
 
 
-async def _servalias_before_edit(ctx, name=None):
+async def _servalias_before_edit(ctx, name=None, delete=False):
     if not _can_edit_servaliases(ctx):
         raise NotAllowed("You do not have permission to edit server aliases. Either __Administrator__ "
                          "Discord permissions or a role named \"Server Aliaser\" or \"Dragonspeaker\" "
                          "is required.")
     await _alias_before_edit(ctx, name)
 
 
-async def _servsnippet_before_edit(ctx, _=None):
+async def _servsnippet_before_edit(ctx, name=None, delete=False):
     if not _can_edit_servaliases(ctx):
         raise NotAllowed("You do not have permission to edit server snippets. Either __Administrator__ "
                          "Discord permissions or a role named \"Server Aliaser\" or \"Dragonspeaker\" "
                          "is required.")
+    await _snippet_before_edit(ctx, name)
+
+
+async def _snippet_before_edit(ctx, name=None, delete=False):
+    if delete:
+        return
+    confirmation = None
+    # special arg checking
+    if not name:
+        raise InvalidArgument("Snippet names can not be empty.")
+    name = name.lower()
+    if name in SPECIAL_ARGS or name in DAMAGE_TYPES or \
+        name in STAT_NAMES or name in STAT_ABBREVIATIONS or \
+        name in SKILL_NAMES or name in STAT_VAR_NAMES or name.startswith('-'):
```

Comment:
Rather than doing this big chain, I'd just define `SPECIAL_ARGS` as the union of all of these iterables (this also prevents us from having to do O(n) iteration over the constant tuples)